developers/weaviate/config-refs/schema/vector-index.md

```diff
@@ -144,6 +147,26 @@ Configure `bq` with these parameters.
 | `rescoreLimit` | integer | -1 | The minimum number of candidates to fetch before rescoring. |
 | `cache` | boolean | `false` | Whether to use the vector cache. |
 
+## Dynamic indexes
+:::warning The `dynamic` indexing feature added in 1.25 is currently in beta.
+
+:::
+
+Using the `dynamic` index will initially create a flat index and once the number of objects exceeds a certain threshold (by default 10,000 objects) it will automatically switch you over to an HNSW index. 
+
+This is only a one-way switch that converts a flat index to a HNSW, the index does not support changing back to a flat index even if the object count goes below the threshold due to deletion. 
+
+The goal of `dynamic` indexing is to shorten latencies during query time at the cost of a larger memory footprint.
+
+### Dynamic index parameters
+| Parameter | Type | Default | Details |
+| :-- | :-- | :-- | :-- |
+| `distance` | string | `cosine` | Distance metric. The metric that measures the distance between two arbitrary vectors. |
+| `hnsw` | VectorIndexConfigHNSW | default HNSW | HNSW index configuration to be used. |
+| `flat` | VectorIndexConfigFlat | default Flat | Flat index configuration to be used. |
+| `threshold` | integer | 10000 | Threshold object count at which `flat` to `hnsw` conversion happens |
+| `vectorCacheMaxObjects`| integer | `1e12` | Yes | Maximum number of objects in the memory cache. By default, this limit is set to one trillion (`1e12`) objects when a new collection is created. For sizing recommendations, see [Vector cache considerations](../../concepts/vector-index.md#vector-cache-considerations). |
+| `quantizer` | string | None | Whether to use `bq` or `pq` quantization. |
```

Comment:
Not sure if I read this lines well. In any case, all vector index config which is relevant to flat or hnsw goes into their individual config. This means that for example you could use quantisation only with one of them. This been said, the quantiser normally goes in the pq/bq fields inside flat or hnsw, not at the root config.